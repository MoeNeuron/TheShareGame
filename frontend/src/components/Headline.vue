<template>
  <div>
    <h1 class="pb-2 mt-4 mb-2 border-bottom" id="headline">{{ name }}</h1>

  </div>
</template>

<script>
export default {
  name: 'Headline',
  props: ['name'],
};
</script>

<style scoped>

</style>
