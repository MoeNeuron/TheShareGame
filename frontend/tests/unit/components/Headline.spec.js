import { shallowMount } from '@vue/test-utils';
import Headline from '@/components/Headline.vue';

describe('Headline.vue', () => {
  it('renders props.name when passed', () => {
    const name = 'A new headline';
    const wrapper = shallowMount(Headline, {
      propsData: { name },
    });
    expect(wrapper.text()).toMatch(name);
  });

  it('matches Snapshot', () => {
    expect(shallowMount(Headline)).toMatchSnapshot();
  });
});
