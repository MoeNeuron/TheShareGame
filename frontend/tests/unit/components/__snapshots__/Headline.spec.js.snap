// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Headline.vue matches Snapshot 1`] = `
<div>
  <h1 id="headline" class="pb-2 mt-4 mb-2 border-bottom"></h1>
</div>
`;
